Apps/GatewayApi/test/unit/Services.Test/UserFeedbackServiceTests.cs

```diff
@@ -105,84 +178,43 @@ private static Mock<ICryptoDelegate> GetCryptoDelegateMock()
-                    ? new()
-                    {
-                        ResultMessage = dbErrorMessage,
-                        ErrorCode = ErrorTranslator.ServiceError(ErrorType.CommunicationInternal, ServiceType.Database),
-                    }
-                    : null,
-                ResourcePayload = dbErrorExists ? null : new() { Id = ratingId, RatingValue = rating.RatingValue, Skip = rating.Skip },
-            };
-
-            Mock<IRatingDelegate> ratingDelegateMock = new();
-            ratingDelegateMock.Setup(s => s.InsertRatingAsync(It.Is<Rating>(r => r.RatingValue == rating.RatingValue && r.Skip == rating.Skip), It.IsAny<CancellationToken>()))
-                .ReturnsAsync(dbResult);
-
-            IUserFeedbackService service = new UserFeedbackService(
+            return new UserFeedbackService(
                 new Mock<ILogger<UserFeedbackService>>().Object,
-                new Mock<IFeedbackDelegate>().Object,
+                feedbackDelegateMock.Object,
                 ratingDelegateMock.Object,
-                new Mock<IUserProfileDelegate>().Object,
-                new Mock<IBackgroundJobClient>().Object,
+                userProfileDelegateMock.Object,
+                backgroundJobClientMock.Object,
                 MappingService,
-                new Mock<IAuthenticationDelegate>().Object);
-
-            return new(service, expected, createRating);
+                authenticationDelegateMock.Object);
         }
 
-        private static CreateUserFeedbackMock SetupCreateUserFeedbackMock()
+        private static IUserFeedbackService SetupCreateRatingMock(DbResult<Rating> dbResult, SubmitRating createRating)
         {
-            const string hdid = "Mocked UserProfileId";
-            const UserLoginClientType defaultClientType = UserLoginClientType.Mobile;
-
-            UserFeedback userFeedback = new()
-            {
-                Id = Guid.NewGuid(),
-                Comment = "Mocked Comment",
-                UserProfileId = hdid,
-                ClientCode = defaultClientType,
-            };
-
-            Feedback feedback = new()
-            {
-                Comment = userFeedback.Comment,
-            };
+            Mock<IRatingDelegate> ratingDelegateMock = new();
+            ratingDelegateMock.Setup(
+                    s => s.InsertRatingAsync(
+                        It.Is<Rating>(r => r.RatingValue == createRating.RatingValue && r.Skip == createRating.Skip),
+                        It.IsAny<CancellationToken>()))
+                .ReturnsAsync(dbResult);
 
-            DbResult<UserFeedback> dbResult = new()
-            {
-                Payload = userFeedback,
-                Status = DbStatusCode.Created,
-            };
+            return GetUserFeedbackService(ratingDelegateMock: ratingDelegateMock);
+        }
 
+        private static IUserFeedbackService SetupCreateUserFeedbackMock(DbResult<UserFeedback> dbResult, UserFeedback createUserFeedback)
```

Comment:
Method name no longer corresponds to the returned type.  I suggest something like `SetupUserFeedbackServiceForCreateUserFeedback()`.

Reply:
done